app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,444 @@
+                thenReturn(LatLng(2.0,1.0,0.0F))
+        //111.19 km real distance, return false if 148306.444306 >  currentLocationSearchRadius
+        NearbyController.currentLocationSearchRadius = 148306.0
+        val isClose = nearbyPresenter?.searchCloseToCurrentLocation()
+        assertFalse(isClose!!)
+    }
+
+    /**
+     * Test if the search is close to current location, when close
+     */
+    @Test
+    fun testSearchCloseToCurrentLocationWhenClose() {
+        whenever(nearbyParentFragmentView.getLastFocusLocation()).
+            thenReturn(com.mapbox.mapboxsdk.geometry.LatLng(1.0,1.0,0.0))
+        whenever(nearbyParentFragmentView.getCameraTarget()).
+            thenReturn(LatLng(2.0,1.0,0.0F))
+        //111.19 km real distance, return false if 148253.333 >  currentLocationSearchRadius
+        NearbyController.currentLocationSearchRadius = 148307.0
+        val isClose = nearbyPresenter?.searchCloseToCurrentLocation()
+        assertTrue(isClose!!)
+    }
+
+    fun expectMapAndListUpdate() {
+        nearbyPresenter.lockUnlockNearby(false)
+        whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
+        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
+    }
+
+    @Test
+    fun testSetActionListeners() {
+        nearbyPresenter.setActionListeners(any())
+        verify(nearbyParentFragmentView).setFABPlusAction(any())
+        verify(nearbyParentFragmentView).setFABRecenterAction(any())
+    }
+
+    @Test
+    fun testBackButtonClickedWhenBottomSheetExpanded() {
+        whenever(nearbyParentFragmentView.isListBottomSheetExpanded()).thenReturn(true)
+        nearbyPresenter.backButtonClicked()
+        verify(nearbyParentFragmentView).listOptionMenuItemClicked()
+    }
+
+    @Test
+    fun testBackButtonClickedWhenDetailsBottomSheetVisible() {
+        whenever(nearbyParentFragmentView.isListBottomSheetExpanded()).thenReturn(false)
+        whenever(nearbyParentFragmentView.isDetailsBottomSheetVisible()).thenReturn(true)
+        nearbyPresenter.backButtonClicked()
+        verify(nearbyParentFragmentView).setBottomSheetDetailsSmaller()
+    }
+
+    @Test
+    fun testBackButtonClickedWhenNoSheetVisible() {
+        whenever(nearbyParentFragmentView.isListBottomSheetExpanded()).thenReturn(false)
+        whenever(nearbyParentFragmentView.isDetailsBottomSheetVisible()).thenReturn(false)
+        nearbyPresenter.backButtonClicked()
+        verify(nearbyParentFragmentView).setTabItemContributions()
+    }
+
+    @Test
+    fun testMarkerUnselected() {
+        nearbyPresenter.markerUnselected()
+        verify(nearbyParentFragmentView).hideBottomSheet();
+    }
+
+    @Test
+    fun testMarkerSelected() {
+        nearbyPresenter.markerSelected(marker)
+        verify(nearbyParentFragmentView).displayBottomSheetWithInfo(marker)
+    }
+
+    @Test
+    fun testOnWikidataEditSuccessful() {
+        nearbyPresenter.onWikidataEditSuccessful()
```

Comment:
you would need to set your expectations first.

It is weird that this test calls 2 methods on `nearbyPresenter` when `onWikidataEditSuccessful` already calls `updateMapAndList`

This goes for the following tests too